OSDOCS-12183: adding how to delete manifests (#83246)
openshift/openshift-docs: modules/microshift-manifests-removal-scenario.adoc

// Module included in the following assemblies:
//
// * microshift//running_applications/microshift-deleting-resource-manifests.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-manifests-removal-scenario_{context}"]
== Manifest removal scenario
Use the following procedure in the data removal scenario to completely delete the resource defined in the kustomize manifests.

> .Procedure

Reviewer: We cannot reuse this procedure for ostree.
It makes sense for rpm systems (with a caveat that both /etc/microshift/manifests.d/delete/ and /usr/lib/microshift/manifests.d/delete can be used, because they're both writable on rpm system).

On ostree systems: those needs to be packaged as RPMs that install to right dir straight away - there's no moving allowed because at runtime the /usr is read-only. But I don't think it's a big deal - if user already has manifests on ostree, they know how to create them, if they want to create "delete" manifests, they just need to use slightly different path.

Author: separate module created for OSTree

. Identify the manifest that needs to be placed in the `delete` subdirectories.
. Use one of the following steps to create the `delete` subdirectories where the manifest is placed for deletion:
* For RPM installation, create the `delete` subdirectory by running the following command:
+
[source,terminal]
----
sudo mkdir -p _<path_of_deletedir>_ <1>
----
<1> Replace `_<path_of_deletedir>_` with the path of delete subdirectory such as `/etc/microshift/manifests.d/delete` or `/etc/microshift/manifests/delete/`.
* For OSTree deployments, create the `delete` subdirectory by running the following command:
+
[source,terminal]
----
sudo mkdir -p _<path_of_deletedir>_ <1>
----

> <1> Replace `_<path_of_deletedir>_` with the path of delete subdirectory such as `/usr/lib/microshift/manifests.d/delete/*` or `/usr/lib/microshift/manifests/delete/`.

Reviewer: As mentioned in comment above - this is incorrect

Author: separate module created for OSTree

+
[IMPORTANT]
====

> For OSTree installation, the `delete` subdirectories are read only. Package the manifest into an RPM, then add them to the blueprint file before moving the manifest into the `delete` subdirectories.

Reviewer: I don't think we can write 
> add them to the blueprint file before moving the manifest into the `delete` subdirectories

Packaging them doesn't let us write to /usr.

```suggestion
For OSTree installation, the `delete` subdirectories are read only. Package the manifest into an RPM, then add them to the blueprint file to install it into correct location.
```

Author: updated and included in the OSTree module

See link:https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/{ocp-version}/html/running_applications/microshift-embedding-apps-tutorial#microshift-adding-app-rpms-to-blueprint_microshift-embedding-apps-tutorial[Adding application RPMs to a blueprint] for the procedure to add an RPM to a blueprint.
====
. Move the manifest file into one of the `delete` subdirectories  under the configured manifests directory by running the following command:
+
[source,terminal]
----
[sudo] mv _<path_of_manifests>_ _<path_of_deletedir>_ <1>
----
<1> Replace `_<path_of_manifests>_` with the path of manifest that has the manifest to be deleted such as `/etc/microshift/manifests.d/` and replace `_<path_of_deletedir>_` with the path of delete subdirectory such as `/etc/microshift/manifests.d/delete` or `/usr/lib/microshift/manifests/delete/`.
. Restart {microshift-short} by running the following command:
+
[source,terminal]
----

> $ sudo systemctl restart microshift

Reviewer: ostree note: because RPM needs to be a part of blueprint, which is turned into an image/deployment, and in order to make that image/deployment active you need to reboot, this step is unnecessary for ostree systems.
